src/spec/index/subscribe.spec.js

```diff
@@ -0,0 +1,338 @@
+						handler: anyFunction
+					}]);
+
+				});
+
+		});
+
+		it('rejects if GroupConsumer constructor throws an error', () => {
+
+			// given
+
+			const
+				handler = sandbox.stub(),
+				config = {};
+
+			mocks.kafka.GroupConsumer.throws(new Error('Constructor error'));
+
+			// when - then
+
+			return expect(subscribe.subscribe({ eventName: 'test', handler, config }))
+				.to.be.rejectedWith('Constructor error');
+
+		});
+
+		it('rejects if GroupConsumer init throws an error', () => {
+
+			// given
+
+			const
+				handler = sandbox.stub(),
+				config = {};
+
+			mocks.kafka.GroupConsumer.prototype.init.throws(new Error('Init error'));
+
+			// when - then
+
+			return expect(subscribe.subscribe({ eventName: 'test', handler, config }))
+				.to.be.rejectedWith('Init error');
+
+		});
+
+		it('rejects if handler is not a function', () => {
+
+			// given
+
+			const
+				handler = {},
+				config = {};
+
+			// when - then
+
+			return expect(subscribe.subscribe({ eventName: 'test', handler, config }))
+				.to.be.rejectedWith('Handler is not a Function.');
+
+		});
+
+	});
+
+	describe('handler', () => {
+
+		let subscribe, mocks;
+
+		beforeEach(() => {
+
+			mocks = {};
+			mocks.kafka = {
+				GroupConsumer: sandbox.stub()
+			};
+
+			mocks.kafka.GroupConsumer.prototype.init = sandbox.stub().resolves();
+			mocks.kafka.GroupConsumer.prototype.commitOffset = sandbox.stub().resolves();
+
+			subscribe = proxyquire('../../lib/index/subscribe', {
+				'no-kafka': mocks.kafka
+			});
+		});
+
+		afterEach(restore);
+
+		it('handler calls underlying one and commits offset', () => {
+
+			// given
+
+			const
+				handler = sandbox.stub().resolves(),
+				config = {};
+
+			// when
+
+			return expect(subscribe.subscribe({ eventName: 'test', handler, config })).to.be.fulfilled
+				.then(() => {
+
+					// then
+
+					const
+						initArgs = mocks.kafka.GroupConsumer.prototype.init.args[0][0][0],
+						messageSet = [{
+							offset: 99,
+							message: {
+								value: Buffer.from('Test')
+							}
+						}],
+						topic = 'test',
+						partition = 0;
+
+					return expect(initArgs.handler(messageSet, topic, partition)).to.be.fulfilled
+						.then(() => {
+
+							calledOnce(handler);
+							calledWith(handler, messageSet[0].message.value);
+
+							calledOnce(mocks.kafka.GroupConsumer.prototype.commitOffset);
+							calledWith(mocks.kafka.GroupConsumer.prototype.commitOffset, {
+								topic: topic,
+								partition: partition,
+								offset: 99
+							});
+
+						});
+
+				});
+
+		});
+
+		it('underlying handler throws then offset is not committed', () => {
+
+			// given
+
+			const
+				handler = sandbox.stub().throws(new Error('Handler error')),
+				config = {};
+
+			// when
+
+			return expect(subscribe.subscribe({ eventName: 'test', handler, config })).to.be.fulfilled
+				.then(() => {
+
+					// then
+
+					const
+						initArgs = mocks.kafka.GroupConsumer.prototype.init.args[0][0][0],
```

Comment:
Personally I would prefer the stub to `yield(messageSet, topic, partition)` and invoke the behaviour during the //when, rather than performing follow-up //whens inside the //then. If that makes any sense!
If you prefer capturing, then go for it.